client/src/components/tables/DataTable.js

```diff
@@ -0,0 +1,206 @@
+import React, { useCallback, useEffect, useState } from 'react';
+
+import { Grid } from '@material-ui/core';
+import { makeStyles } from '@material-ui/core/styles';
+
+import { Table, Button } from '../../components/generic';
+
+import dayjs from 'dayjs';
+
+import { sortObjects } from '../../utils';
+
+export const useTableStyles = makeStyles((theme) => ({
+  rootItem: {
+    paddingLeft: theme.spacing(2),
+    paddingRight: theme.spacing(2),
+    paddingTop: theme.spacing(1),
+    paddingBottom: theme.spacing(1),
+  },
+  tableItem: {
+    paddingTop: theme.spacing(4),
+    paddingRight: theme.spacing(2),
+    paddingBottom: theme.spacing(4),
+    paddingLeft: theme.spacing(2),
+  },
+  filterLabel: {
+    color: theme.palette.gray.dark,
+    fontWeight: 700,
+  },
+  actionMenuPaper: {
+    minWidth: '220px',
+  },
+  menuItem: {
+    padding: '.75rem',
+    fontSize: '17px',
+  },
+}));
+
+/**
+ * @typedef {Object} modalComponentProps
```

Comment:
Is this comment in the right place?

Reply:
Are you referring to it being below `useTableStyles`? The intention there was that since all of these types are used by the component itself, I'd define it after the semi-separate style function, but it is a bit odd there I suppose. Moving these type definitions inside the function wouldn't work well though, if that's what you mean.

Either way, hopefully at some point down the road we can start supporting actual TS and this will all get a lot cleaner.

Reply:
Understood! Yeah, it could potentially be a file doc block, but... Regardless nitpick at this point.
Thank you for the explanation.